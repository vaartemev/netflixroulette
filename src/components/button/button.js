import React from 'react';
import PropTypes from 'prop-types';
import './button.scss';

export const Button = ({ classNames }) => {
  return <div className={classNames}>Search</div>;
};

Button.propTypes = {
  classNames: PropTypes.string
};
