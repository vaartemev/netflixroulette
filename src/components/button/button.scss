.button {
	display: inline-block;
	border-radius: 3px;
	transition: .3s;
	cursor: pointer;

	&__search {
		text-transform: uppercase;
		position: absolute;
		right: 0;
		top: 50%;
		transform: translateY(-50%);
		height: 40px;
		line-height: 40px;
		font-size: 20px;
		padding: 0 80px;
		background-color: #f55263;
		color: #ffffff;

		&-link {
			padding: 0 30px;
			background-color: #ffffff;
			color: #f55263;
			font-size: 14px;
			height: 30px;
			line-height: 30px;
			right: 70px;
			top: 20px;
			transform: none;
			&:hover {
				background-color: #f55263;
				color: #ffffff;
			}
		}
	}

	&__sort {
		color: #656565;
		display: inline-block;
		margin-right: 10px;
		transition: .3s;
		cursor: pointer;
		background: transparent;

		&:hover {
			color: #f55263;
		}

		&-active {
			color: #f55263;
		}
	}

	&__filter {
		margin-right: 10px;
		height: 30px;
		line-height: 30px;
		padding: 0 10px;
		color: #ffffff;
		background-color: #555555;
		text-transform: uppercase;

		&-active {
			background-color: #f55263;
		}
	}
}