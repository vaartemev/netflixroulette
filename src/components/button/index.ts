export { Button, ButtonType } from './button';
