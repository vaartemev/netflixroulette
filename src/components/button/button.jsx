import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './button.scss';

export const ButtonType = {
  search: 'search',
  searchLink: 'search-link',
  filter: 'filter',
  sort: 'sort',
};

export const Button = ({
  text,
  checked,
  handleOnClick,
  type = ButtonType.sort,
}) => {
  const resultClass = classNames('button', {
    'button__filter-active': checked && type === ButtonType.filter,
    'button__sort-active': checked && type === ButtonType.sort,
    button__sort: type === ButtonType.sort,
    button__search: type === ButtonType.search,
    'button__search button__search-link': type === ButtonType.searchLink,
    button__filter: type === ButtonType.filter,
  });

  return (
    <button className={resultClass} type="button" onClick={handleOnClick}>
      {text}
    </button>
  );
};

Button.propTypes = {
  text: PropTypes.string,
  checked: PropTypes.bool,
  handleOnClick: PropTypes.func,
  type: PropTypes.string,
};
