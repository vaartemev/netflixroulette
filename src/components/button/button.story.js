import React from 'react';
import { storiesOf } from '@storybook/react';
import { Button, ButtonType } from './button';
import { RadioButton } from '../../containers/radioButton';

storiesOf('Button', module)
  .add('Button', () => <Button type={ButtonType.search} text="Search" />)
  .add('Button', () => <Button type={ButtonType.searchLink} text="Search" />)
  .add('Radio Filter', () => (
    <RadioButton
      type={ButtonType.filter}
      array={[{ key: 'title' }, { key: 'title2' }]}
    />
  ))
  .add('Radio Sort', () => (
    <RadioButton
      type={ButtonType.sort}
      array={[{ key: 'title' }, { key: 'title2' }]}
    />
  ));
