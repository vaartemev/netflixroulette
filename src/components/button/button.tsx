import React from 'react';
import Link from 'next/link';
import classNames from 'classnames';
import './button.scss';

export const ButtonType = {
  search: 'search',
  searchLink: 'search-link',
  filter: 'filter',
  sort: 'sort',
};

interface Props {
  text: string;
  checked?: boolean;
  handleOnClick?: () => void;
  type?: string;
  href: URL;
}

export const Button = ({
  text,
  checked,
  handleOnClick,
  type = ButtonType.sort,
  href,
}: Props) => {
  const resultClass = classNames('button', {
    'button__filter-active': checked && type === ButtonType.filter,
    'button__sort-active': checked && type === ButtonType.sort,
    button__sort: type === ButtonType.sort,
    button__search: type === ButtonType.search,
    'button__search button__search-link': type === ButtonType.searchLink,
    button__filter: type === ButtonType.filter,
  });

  return (
    <Link href={href}>
      <button className={resultClass} type="button" onClick={handleOnClick}>
        {text}
      </button>
    </Link>
  );
};
