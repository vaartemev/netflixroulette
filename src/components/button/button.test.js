import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import { Button } from './button';

describe('<Button />', () => {
  it('Should render with Hello!', () => {
    const output = shallow(<Button text="Hello" />);
    expect(output.find('button').text()).toEqual('Hello');
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Test click event on button', () => {
    const handleOnClick = sinon.spy();
    const output = shallow(<Button handleOnClick={handleOnClick} />);
    output.find('button').simulate('click');
    expect(handleOnClick).toHaveProperty('callCount', 1);
  });
});
