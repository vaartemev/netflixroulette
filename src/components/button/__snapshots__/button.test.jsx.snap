// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render with Hello! 1`] = `
<button
  className="button button__sort"
  type="button"
>
  Hello
</button>
`;
